src/itemlistformaction.cpp

```diff
@@ -491,10 +491,23 @@ void ItemListFormAction::process_operation(Operation op,
 			if (cfg->get_configvalue_as_bool(
 				    "markfeedread-jumps-to-next-unread")) {
 				process_operation(OP_NEXTUNREAD);
-			} else {
-				LOG(Level::DEBUG,
-					"ItemListFormAction::reset itempos");
-				f->set("itempos", "0");
+			} else { // reposition to first/last item
+				std::string sortorder =
+					cfg->get_configvalue("article-sort-order");
+
+				if (sortorder.find("date-desc") != std::string::npos) {
```

Comment:
`find` implies that we're matching a part of the string, but here, we're checking the whole of it; I think `sortorder == "date-desc"` will be sufficient.

Reply:
even better :)